internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -74,3 +80,129 @@ func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationR
+	default:
+		return fmt.Errorf("unsupported monitoring management state: %s", dsci.Spec.Monitoring.ManagementState)
+	}
+}
+
+// createOrUpdateInstrumentationCR creates or updates the OpenTelemetry Instrumentation CR using unstructured resources
+func createOrUpdateInstrumentationCR(ctx context.Context, cli client.Client, name, namespace string, dsci *dsciv1.DSCInitialization) error {
+	// Set default sample ratio if not specified
+	sampleRatio := "0.1"
+	if dsci.Spec.Monitoring.Traces != nil && dsci.Spec.Monitoring.Traces.SampleRatio != nil {
+		sampleRatio = *dsci.Spec.Monitoring.Traces.SampleRatio
+	}
+
+	// Define the GVK for Instrumentation
+	gvk := schema.GroupVersionKind{
+		Group:   "opentelemetry.io",
+		Version: "v1alpha1",
+		Kind:    "Instrumentation",
+	}
+
+	// Create the unstructured instrumentation object
+	instrumentation := &unstructured.Unstructured{}
+	instrumentation.SetGroupVersionKind(gvk)
+	instrumentation.SetName(name)
+	instrumentation.SetNamespace(namespace)
+
+	// Set labels for OpenDataHub management
+	labels := map[string]string{
+		"opendatahub.io/managed":          "true",
+		"platform.opendatahub.io/part-of": "opendatahub",
+	}
+	instrumentation.SetLabels(labels)
+
+	// Configure the instrumentation spec using unstructured format
+	spec := map[string]interface{}{
+		"exporter": map[string]interface{}{
+			"endpoint": "http://jaeger-collector.istio-system.svc.cluster.local:14268/api/traces",
+		},
+		"sampler": map[string]interface{}{
+			"type":     "traceidratio",
+			"argument": sampleRatio,
+		},
+		"python": map[string]interface{}{
+			"image": "ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-python:latest",
+		},
+		"java": map[string]interface{}{
+			"image": "ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-java:latest",
+		},
+		"nodejs": map[string]interface{}{
+			"image": "ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-nodejs:latest",
+		},
+		"dotnet": map[string]interface{}{
+			"image": "ghcr.io/open-telemetry/opentelemetry-operator/autoinstrumentation-dotnet:latest",
+		},
+	}
+
+	if err := unstructured.SetNestedMap(instrumentation.Object, spec, "spec"); err != nil {
+		return fmt.Errorf("failed to set instrumentation spec: %w", err)
+	}
+
+	// Try to get existing instrumentation
+	existing := &unstructured.Unstructured{}
+	existing.SetGroupVersionKind(gvk)
+	err := cli.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, existing)
+
+	if k8serr.IsNotFound(err) {
+		// Create new instrumentation CR
+		return cli.Create(ctx, instrumentation)
+	} else if err != nil {
+		return fmt.Errorf("failed to get existing instrumentation CR: %w", err)
+	}
+
+	// Update existing instrumentation CR - copy the spec and labels
+	if err := unstructured.SetNestedMap(existing.Object, spec, "spec"); err != nil {
+		return fmt.Errorf("failed to update instrumentation spec: %w", err)
+	}
+	existing.SetLabels(labels)
+
+	return cli.Update(ctx, existing)
```

Comment:
instead of get+update, can we use server side apply ? just use `rr.AddResource(...)`